01_fib/lexual/main_test.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+)
+
+func TestFibPositive(t *testing.T) {
+	numbers := []string{
+		"1",
+		"1",
+		"2",
+		"3",
+		"5",
+		"8",
+		"13",
+	}
+	runTest(t, 7, numbers)
```

Comment:
I would avoid having a helper function here to run the test. What is normal convention is to have the test comparison logic and `testing.T` method calls done inside the base test. 

I can see where you are coming from with this approach, thinking to abstract out common code into a function, but perhaps you should think in this case, maybe I could structure my test differently to avoid duplicating this code twice. This brings me to incorporating test cases. Perhaps you could combine the test cases into a data structure and iterate over that data structure in the one test.

It is idiomatic in go to have multiple cases for the one function in a single test, rather than multiple tests for each case on a function.

See [Testing Cases](https://goplay.space/#Zy2ta-z8vrO) for inspiration.

Reply:
see latest push